lib/classes/PluginManager.test.js

```diff
@@ -377,7 +379,7 @@ describe('PluginManager', () => {
     let getCommandsStub;
 
     beforeEach(() => {
-      writeFileStub = sinon.stub().resolves();
+      writeFileStub = sinon.stub().returns(BbPromise.resolve());
```

Comment:
@pmuens This is critical. Someone added a `stub().resolves()` here, but this is not available in sinon 1.x which is used in Serverless. I fixed it by using `.returns(...)`. We should immediately update to sinon 2.x because people are already used to it, and as long as the unit tests are completely broken, noone will ever see the issues and the codebase deteriorates and gets bad.

Reply:
@HyperBrain Really, really nice catch 👍

I used `resolves()` for the last couple of months w/o noticing that we're running such an old `sinon` version! Yes, we should definitely update it ASAP!

Reply:
sinon v3.x has been already released.
http://sinonjs.org/releases/index.html

We should update to follow the migration guide. 
Here are the steps.
http://sinonjs.org/releases/v3.2.1/migrating-to-2.0/
http://sinonjs.org/releases/v3.2.1/migrating-to-3.0/

Reply:
Thanks for the links @horike37 👍

I just figured that this `resolves` should work since we use `sinon-bluebird`. However we should update `sinon` ASAP (we're currently on v1 and v3 is the latest one).

I started the update in https://github.com/serverless/serverless/pull/4200 (but it's currently on hold since we need a way to use Bluebird rather than the native Promise implementation).

Reply:
@pmuens `sinon-bluebird` is deprecated with version 2.

Regarding the promise type used by sinon. It's quite easy to set the implementation:
`sinon.usingPromise(BbPromise.Promise)`